database/database.go

```diff
@@ -0,0 +1,91 @@
+package database
+
+import (
+	"dashboard/model"
+	"database/sql"
+	"log"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+var db *sql.DB
+var err error
+
+//InitialiseDb to start db server
+func InitialiseDb() {
+	db, err = sql.Open("mysql", "root:mysql1234@tcp(localhost:3306)/dashboard")
+	if err != nil {
+		log.Panic(err.Error())
+	}
+}
+
+//CloseDb to close DB connection
+func CloseDb() {
+	db.Close()
+}
+
+//InsertUser func for sign_up
+func InsertUser(user *model.UserProfile) string {
+
+	stmtOut, err := db.Prepare("SELECT username FROM users WHERE username  = ? ")
+	if err != nil {
+		panic(err.Error())
+	}
+	defer stmtOut.Close()
+	var username string
+	var result string
+	err = stmtOut.QueryRow(user.Username).Scan(&username)
+	if err != nil {
+		hash, err := bcrypt.GenerateFromPassword([]byte(user.Password), 4)
+		if err != nil {
+			panic(err.Error())
+		}
+		user.Password = string(hash)
+
+		stmt, err := db.Prepare("INSERT INTO users(Username,FirstName,LastName,Password,Gender,Country,Age,Email) VALUES(?,?,?,?,?,?,?,?)")
+		if err != nil {
+			log.Panic(err)
+		}
+		defer stmt.Close()
+
+		_, err = stmt.Exec(user.Username, user.FirstName, user.LastName, user.Password, user.Gender, user.Country, user.Age, user.Email)
+		if err != nil {
+			log.Panic(err)
+		}
+		result = "Registration Successful"
+	}
+	result = "Already Registered Username"

```

Comment:
Instead of returning a string, return an error from InsertUserProfile function. GetUserProfileByUsername function should return valid User Profile object or nil and an error (which would be nil, if there is no error)